spring-integration-gcp/src/main/java/org/springframework/integration/gcp/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -56,8 +58,10 @@ public PubSubInboundChannelAdapter(PubSubOperations pubSubTemplate,
 
 		StringMessageConverter stringMessageConverter = new StringMessageConverter();
 		stringMessageConverter.setSerializedPayloadClass(String.class);
+		List<MessageConverter> messageConverters = new ArrayList<>();
+		messageConverters.add(stringMessageConverter);
 		getMessagingTemplate().setMessageConverter(
-				new CompositeMessageConverter(ImmutableList.of(stringMessageConverter)));
+				new CompositeMessageConverter(messageConverters));
```

Comment:
Why do you wrap to `CompositeMessageConverter` at all if we have only one `StringMessageConverter` in the list?

Reply:
I was thinking we might want to add more in the future, but this is the classic case of YAGNI.
If we need more converters in the future, we can add the `CompositeMessageConverter` at that time.

Reply:
I think we don't. 
If end-user needs, he can simply inject this `CompositeMessageConverter` thanks to our setter on the matter.